### ISSUE TITLE
Wrong creation of collections.deque parameter

### ISSUE THREAD

**Issue by wardvandriessche-centrica** [pydantic/pydantic#5452]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

When creating a class from a Pydantic BaseModel with a `collections.deque` parameter, the `maxlen` parameter is not propagated through when an instance is initialized with an existing deque object.

### Example Code

```Python
from pydantic import BaseModel
from typing import Deque
from collections import deque
 
class PastConsumption(BaseModel):
consumption: Deque[int] = deque(maxlen=5) 

a = PastConsumption()

print(a.consumption.maxlen)   # -> 5 as output, as expected

b = PastConsumption(consumption=deque(maxlen=8))

print(b.consumption.maxlen)    # -> "None", BUG
```

### Python, Pydantic & OS Version

```Text
❯ python -c "import pydantic.utils; print(pydantic.utils.version_info())"

             pydantic version: 1.10.7
            pydantic compiled: True
                 install path: <hidden>
               python version: 3.10.9 (main, Jan 10 2023, 15:38:42) [Clang 14.0.0 (clang-1400.0.29.202)]
                     platform: macOS-13.2.1-arm64-arm-64bit
     optional deps. installed: ['typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by hramezani** [comment]
Thanks @wardvandriessche-centrica for reporting.
I can confirm this on both V1 and V2. Don't have any clue what is the reason here.
@samuelcolvin @dmontagu  any idea?

**Comment by dmontagu** [comment]
I think I've got a fix for v2, will open PR shortly.

**Comment by dmontagu** [comment]
Update: the (v2) fix for this is now available in the latest v2 alpha release `v2.0a3`

**Comment by hramezani** [comment]
As @dmontagu mentioned this is fixed in `V2` and is available in `v2.0a3`
